libbeat/common/url.go

```diff
@@ -83,3 +83,19 @@ func EncodeURLParams(url string, params url.Values) string {
 
 	return strings.Join([]string{url, "?", params.Encode()}, "")
 }
+
+// ParseURL tries to parse a URL and return the parsed result.
+func ParseURL(raw string) (*url.URL, error) {
```

Comment:
The default scheme should be configurable as well.

Reply:
I tried to implement this in 9b5fc194d. Please let me know what you think.